fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -184,192 +213,593 @@ public void executeWithoutPassword(BaseRequest request, BaseResponse response) t
+                                                         String label,
+                                                         LoadJobSourceType sourceType,
+                                                         long timeoutMillis) throws UserException {
+            if (!LoadJobSourceType.BYPASS_WRITE.equals(sourceType)) {
+                return null;
+            }
 
-        if (Strings.isNullOrEmpty(dbName)) {
-            throw new UserException("No database selected.");
+            long dbId = db.getId();
+            long tableId = table.getId();
+
+            TxnCoordinator coordinator = new TxnCoordinator(TxnSourceType.FE, HOST_NAME);
+            GlobalTransactionMgr globalTxnMgr = GlobalStateMgr.getCurrentState().getGlobalTransactionMgr();
+            long txnId = globalTxnMgr.beginTransaction(
+                    dbId, Lists.newArrayList(tableId), label, coordinator, sourceType, timeoutMillis);
+            TransactionResult result = new TransactionResult();
+            result.addResultEntry(TransactionResult.TXN_ID_KEY, txnId);
+            result.addResultEntry(TransactionResult.LABEL_KEY, label);
+            return result;
         }
-        if (Strings.isNullOrEmpty(label)) {
-            throw new UserException("empty label.");
+
+        private TransactionResult handlePrepareTransaction(Database db,
+                                                           String label,
+                                                           List<TabletCommitInfo> committedTablets,
+                                                           List<TabletFailInfo> failedTablets) throws UserException {
+            long dbId = db.getId();
+            GlobalTransactionMgr globalTxnMgr = GlobalStateMgr.getCurrentState().getGlobalTransactionMgr();
+            TransactionState txnState = globalTxnMgr.getLabelTransactionState(dbId, label);
+            if (null == txnState) {
+                throw new UserException(String.format("No transaction found by label %s", label));
+            }
+
+            if (!LoadJobSourceType.BYPASS_WRITE.equals(txnState.getSourceType())) {
+                return null;
+            }
+
+            long txnId = txnState.getTransactionId();
+            TransactionStatus txnStatus = txnState.getTransactionStatus();
+            TransactionResult result = new TransactionResult();
+            switch (txnStatus) {
+                case PREPARE:
+                    globalTxnMgr.prepareTransaction(
+                            dbId, txnId, committedTablets, failedTablets, new MiniLoadTxnCommitAttachment());
+                    result.addResultEntry(TransactionResult.TXN_ID_KEY, txnId);
+                    result.addResultEntry(TransactionResult.LABEL_KEY, label);
+                    break;
+                case COMMITTED:
+                case VISIBLE:
+                    result.setOKMsg(String.format("Transaction %s has already committed, label is %s", txnId, label));
```

Comment:
```suggestion
                    result.setOKMsg(String.format("Transaction %d has already {real state}, label is %s", txnId, label));
```

Reply:
PREPARED status should return ok

Reply:
Yeah, the`PREPARED` status should indeed return ok, and i have fixed it